engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessInstanceElementActivatingApplier.java

```diff
@@ -150,6 +151,9 @@ private void decrementActiveSequenceFlow(
       case EVENT_SUB_PROCESS:
         decrementEventSubProcessSequenceFlow(value, flowScopeInstance);
         break;
+      case END_EVENT:
+        decrementEndEventSequenceFlow(value, flowScopeInstance);
```

Comment:
:wrench: We could move the logic to the `completed` event applier instead. 

The parallel elements in the flow scope are terminated when the terminate end event is completed. So, it may fit better to reset the sequence flow counter on completed.